ANMN/burst_averaged_product/burst_average_product.py

```diff
@@ -0,0 +1,388 @@
+    # shutil.rmtree(tempdir)
+    file_path = MooringFileClassifier.dest_path(input_netcdf_file_path)
+    return file_path
+
+def get_burst_duration(netcdf_file_obj):
+    """
+    returns the burst duration as float defined in the NetCDF file if exists
+    """
+    burst_duration  = getattr(netcdf_file_obj, 'instrument_burst_duration', 60.0)
+    return burst_duration
+
+def min_bad_qc_flag(varname):
+    """returns the min IMOS IODE qc flag value to get rid of per variable"""
+    default_min_qc_flag_value = 3
+    dict_var = {
+        'CHLF':     4,
+        'CHLR':     4,
+        'CHLU':     4,
+        'CPHL':     4,
+        'DOX1_2':   3,
+        'DOX2':     3,
+        'FLU2':     4,
+        'FLNTU':    4,
+        'PRES_REL': 4,
+        'PSAL':     4,
+        'TEMP':     4,
+        'TURB':     4,
+    }
+
+    if varname in dict_var.keys():
+        return dict_var[varname]
+    else:
+        return default_min_qc_flag_value
+
+def create_burst_average_var(netcdf_file_obj):
+    """
+    create burst data from all vars available in netcdf
+    """
+    time_values              = get_time_val(netcdf_file_obj)
+    varname_to_burst_average = list_var_to_average(netcdf_file_obj)
+    burst_duration           = get_burst_duration(netcdf_file_obj)
+    burst_vars               = {}
+
+    for varname in varname_to_burst_average:
+        var_values, var_qc_flag_exclusion = get_var_val_var_qc_exclusion(
+            netcdf_file_obj, varname, min_bad_qc_flag(varname)
+        )
+        burst_vars[varname] = list(
+            burst_average_data(time_values, var_values, burst_duration, var_qc_flag_exclusion)
+        )
+
+    burst_vars = clean_burst_vars(burst_vars)
+    return burst_vars
+
+def clean_burst_vars(burst_vars):
+    """
+    For every burst var created, look for the first index of non NaN value. The
+    lower value will be the one kept, and the new start index of each burst variable
+    including the TIME.
+    """
+    min_index = None
+
+    for var in burst_vars.keys():
+        var_mean_burst = burst_vars[var][1] # first non TIME product
+        if not np.isnan(var_mean_burst).all():
+            min_index_var  = next(x for x, y in enumerate(var_mean_burst) if not isnan(y))
+        else:
+            min_index_var = 0
+
+        if min_index == None:
+            min_index = min_index_var
+        else:
+            min_index = min(min_index, min_index_var)
+
+    # remove the first common NaN to all variables
+    for var in burst_vars.keys():
+        for product in range(len(burst_vars[var])):
+            burst_vars[var][product] = burst_vars[var][product][min_index:-1]
+
+    # look for last non NaN. we just have to reverse the list
+    max_non_nan_idx = None
+    for var in burst_vars.keys():
+        var_mean_burst       = burst_vars[var][1] # first non TIME product
+        var_mean_burst       = pd.Series(var_mean_burst) # create a pandas series to find easily last non nan index
```

Comment:
You can get the last non-nan index like this `max( np.where(np.isfinite(var_mean_burst))[0] )`
Otherwise you need to add `pandas` to the requirements.txt file for the repo.